CONTRIBUTING.md

```diff
@@ -0,0 +1,269 @@
+   interface files if your project doesn't already follow another
+   convention
+-  [SF.5](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Rs-consistency)
+   A ``.cpp`` file must include the ``.hpp`` file(s) that defines its
+   interface
+-  [SF.7](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Rs-using-directive)
+   Don't put a global ``using``-directive in a header file
+-  [SF.8](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Rs-guards)
+   Use ``#include`` guards for all ``.hpp`` files
+-  [SF.21](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Rs-unnamed)
+   Don't use an unnamed (anonymous) ``namespace`` in a header
+-  [SL.10](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Rsl-arrays)
+   Prefer using ``std::array`` or ``std::vector`` instead of a C array
+-  [C.9](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Rc-private)
+   Minimize the exposure of class members
+-  [F.3](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Rf-single)
+   Keep functions short and simple
+-  [F.21](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Rf-out-multi)
+   To return multiple 'out' values, prefer returning a ``std::tuple``
+-  [R.1](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Rr-raii)
+   Manage resources automatically using RAII (this includes
+   ``std::unique_ptr`` & ``std::shared_ptr``)
+-  [ES.11](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Res-auto)
+   Use ``auto`` to avoid redundant repetition of type names
+-  [ES.20](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Res-always)
+   Always initialize an object
+-  [ES.23](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Res-list)
+   Prefer the ``{}`` initializer syntax
+-  [CP.1](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#S-concurrency)
+   Assume that your code will run as part of a multi-threaded program
+-  [I.2](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Ri-global)
+   Avoid global variables
+
+## Pull Request Guidelines ##
+Our code contribution guidelines closely follows the model of [GitHub Pull-Requests](https://help.github.com/articles/using-pull-requests).
+The rocWMMA repository follows a workflow which dictates a ``/master`` branch where releases are cut, and a ``/develop`` branch which serves as
+an integration branch for new code.
+
+No changes are allowed to be directly committed to the develop branch of the rocWMMA repository. All authors are required to develop their
+change sets on a separate branch and then create a pull request to merge their changes into the develop branch. When you create a pull
```

Comment:
"all authors are required to develop their change sets on a separate branch (preferably in your own fork), and then create a pull request targeting the develop branch of the upstream repository."